wipac_dev_tools/setup_tools.py

```diff
@@ -0,0 +1,184 @@
+                f"( {sys.version_info} < {self.py_min} )"
+            )
+            sys.exit(1)
+        elif sys.version_info > self.py_max:
+            print(
+                f"WARNING: {self.name} does not officially support Python "
+                f"{sys.version_info[0]}.{sys.version_info[1]}+ "
+                f"( {sys.version_info} > {self.py_max} )"
+            )
+
+    def _find_version(self, init_version_dir: str) -> str:
+        """Grab the package's version string."""
+        fpath = os.path.join(self.here, self.name, init_version_dir, "__init__.py")
+        with open(fpath) as init_f:
+            for line in init_f.readlines():
+                if "__version__" in line:
+                    # grab "X.Y.Z" from "__version__ = 'X.Y.Z'" (quote-style insensitive)
+                    return line.replace('"', "'").split("=")[-1].split("'")[1]
+
+        raise Exception(f"cannot find `__version__` string in '{fpath}'")
+
+    def get_install_requires(self, allow_git_urls: bool = True) -> List[str]:
+        """Get the `install_requires` list."""
+
+        def convert(req: str) -> str:
+            # GitHub Packages
+            if "github.com" in req:
+                if not allow_git_urls:
+                    raise Exception(
+                        "This package cannot contain any git/github url dependencies. "
+                        "This is to prevent any circular dependencies. "
+                        f"The culprit: {req} from {self.requirements_path}"
+                    )
+                pat = r"^git\+(?P<url>https://github\.com/[^/]+/[^/]+)@(?P<tag>(v)?\d+\.\d+\.\d+)#egg=(?P<package>\w+)$"
+                re_match = re.match(pat, req)
+                if not re_match:
+                    raise Exception(
+                        f"from {self.requirements_path}: "
+                        f"pip-install git-package url is not in standardized format {pat} ({req})"
+                    )
+                groups = re_match.groupdict()
+                # point right to .zip (https://stackoverflow.com/a/56635563/13156561)
+                return f'{groups["package"]} @ {groups["url"]}/archive/refs/tags/{groups["tag"]}.zip'
+            # PyPI Packages
+            else:
+                return req.replace("==", ">=")
+
+        return [convert(m) for m in open(self.requirements_path).read().splitlines()]
+
+    def _get_py_classifiers(self) -> List[str]:
+        """NOTE: Will not work after the '3.* -> 4.0'-transition."""
+        if self.py_min[0] < 3:
+            raise Exception("Python-classifier automation does not work for python <3.")
+        if self.py_max[0] >= 4:
+            raise Exception("Python-classifier automation does not work for python 4+.")
+
+        return [
+            f"Programming Language :: Python :: 3.{r}"
+            for r in range(self.py_min[1], self.py_max[1] + 1)
+        ]
+
+    def _get_development_status(self) -> str:
```

Comment:
Would it be worth making the return type `List[str]` so it parallels the other classifier helper?

Reply:
I don't think it would since there will always be a single development status. I agree it makes the list building clunkier.